bokehjs/src/lib/models/annotations/slope.ts

```diff
@@ -0,0 +1,120 @@
+import {Annotation, AnnotationView} from "./annotation"
+import {LineMixinScalar} from "core/property_mixins"
+import {Line} from "core/visuals"
+import {Color} from "core/types"
+import * as p from "core/properties"
+
+export class SlopeView extends AnnotationView {
+  model: Slope
+  visuals: Slope.Visuals
+
+  initialize(options: any): void {
+    super.initialize(options)
+  }
+
+  connect_signals(): void {
+    super.connect_signals()
+    this.connect(this.model.change, () => this.plot_view.request_render())
+    this.connect(this.model.properties.gradient.change, () => this.plot_view.request_render())
+    this.connect(this.model.properties.y_intercept.change, () => this.plot_view.request_render())
+  }
+
+  render(): void {
+    if (!this.model.visible)
+      return
+    this._draw_span()
+  }
+
+  protected _draw_span(): void {
+    const gradient = this.model.gradient
+    const y_intercept = this.model.y_intercept
+    if(gradient == null || y_intercept == null){
+      return
+    }
+
+    const {frame} = this.plot_view
+
+    const xscale = frame.xscales[this.model.x_range_name]
+    const yscale = frame.yscales[this.model.y_range_name]
+
+    let sy_start, sy_end, y_start, y_end
+    sy_start = frame._top.value
+    sy_end = sy_start + frame._height.value
+
+    y_start = yscale.invert(sy_start)
+    y_end = yscale.invert(sy_end)
+
+    let sx_start, sx_end, x_start, x_end
+    x_start = (y_start - y_intercept) / gradient
+    x_end = (y_end - y_intercept) / gradient
+
+    sx_start = xscale.compute(x_start)
+    sx_end = xscale.compute(x_end)
```

Comment:
None of these eight values appear to be re-assigned, so they can all eight just be `const` instead of `let`